pkg/core/resources/apis/mesh/validators.go

```diff
@@ -234,14 +239,14 @@ func ValidateHostname(path validators.PathBuilder, hostname string) validators.V
 	}
 
 	if strings.HasPrefix(hostname, "*.") {
-		if !DomainRegexp.MatchString(strings.TrimPrefix(hostname, "*.")) {
+		if !DNS1123SubdomainRegexp.MatchString(strings.TrimPrefix(hostname, "*.")) {
 			err.AddViolationAt(path, "invalid wildcard domain")
 		}
 
 		return err
 	}
 
-	if !DomainRegexp.MatchString(hostname) {
+	if !DNS1123SubdomainRegexp.MatchString(hostname) {
```

Comment:
The PR description mentions only adding deprecations, but this file contains an actual change in validation. What's the motivation to change the way MeshGateway validates hostnames?

Reply:
I gonna move it to another PR